storage/badger/procedure/index.go

```diff
@@ -35,10 +39,17 @@ func RetrieveIndex(blockID flow.Identifier, index *flow.Index) func(tx *badger.T
 		if err != nil {
 			return fmt.Errorf("could not retrieve seal index: %w", err)
 		}
+		var receiptIDs []flow.Identifier
+		err = operation.LookupPayloadReceipts(blockID, &receiptIDs)(tx)
+		if err != nil {
+			return fmt.Errorf("could not retrieve receipts index: %w", err)
+		}
 
 		*index = flow.Index{
+			NodeIDs:       nil,
 			CollectionIDs: collIDs,
 			SealIDs:       sealIDs,
+			ReceiptIDs:    receiptIDs,
```

Comment:
The ReceiptIDs field was not included in the index, which causes the Receipts field of the block payload to be empty when it's retrieved from database. 

It wasn't a problem, until we start including receipts in the payload. And also our cache layer in the storage will cache the stored payload, which contains the receipts, but when the cache is gone after a restart, the payload returned from the storage didn't have the receipts, and caused problem. 

I've added test cases to verify the same block is able to be retrieved from both the storage before and after a restart.